### DISCUSSION TITLE
col LIKE {{my_param}}

### DISCUSSION THREAD

**Discussion by vivek7405** [tinybirdco/web-analytics-starter-kit#72] [Q&A] [answered]
I am trying to perform a search query with arbitrary string search operator (%):

{% if defined(domain) %}
  and href like '%{{domain}}%'
{% end %}

Note that href is a column in the datasource and domain is a dynamic param.

The above query doesn't work and throws the following error:
DB::Exception: Syntax error: failed at position 277 ('https') (line 17, col 26): https://demo.page.careers/'%' group by pathname, href order by visits desc limit toInt32('0'),toInt32('8'). Expected one of: token, DoubleColon, OR, AND, IS NULL, IS NOT NULL, BETWEEN, NOT BETWEEN, LIKE, ILIKE, NOT LIKE, NOT ILIKE, REGEXP, IN, NOT IN, GLOBAL IN, GLOBAL NOT IN, MOD, DIV, alias, AS, GROUP BY, WITH, HAVING, WINDOW, ORDER BY, LIMIT, OFFSET, FETCH, SETTINGS, UNION, EXCEPT, INTERSECT, INTO OUTFILE, FORMAT, end of query

Though, if I run the following query without arbitrary string search operator (%), it works fine:
{% if defined(domain) %}
  and href like {{domain}}
{% end %}

I am trying to run this under the WHERE clause of the query. I want to perform an arbitrary string search.

Please help me out, I am stuck!! I didn't find any help from tinybird or clickhouse docs.

Thanks in advance for any help.

**Comment by alejandromav** [comment]
Hey @vivek7405, could you please share a sample request URL including that parameter? Thanks

**Comment by Chaffelson** [comment, marked as answer]
Hi Vivek,
I think probably you are running into an issue where the templating replacement of `{{domain}}` is not happening, the way to do this is to use a concat.

`CONCAT('%', {{ String(domain) }}, '%')`

**Comment by vivek7405** [comment]
Yes, I tried concat and it works fine.

Thanks,
Vivek
